src/InterNations/Sniffs/Syntax/MethodTypeHintsSniff.php

```diff
@@ -0,0 +1,324 @@
+                $paramCount++;
+
+                $typeHintPtr = $file->findPrevious([T_WHITESPACE, T_ELLIPSIS], ($i - 1), null, true);
+
+                // Check for no param
+                if (isset(self::$whitelist[$methodName]) && self::$whitelist[$methodName][0] === null) {
+                    $error = sprintf(
+                        'Expected no arguments for this magic method "%1$s::%2$s" found "%3$s"',
+                        $className,
+                        $methodName,
+                        $tokens[$i]['content']
+                    );
+
+                    $file->addError($error, $typeHintPtr, 'WrongArgumentForMethod');
+
+                    continue;
+                }
+
+                // Check for mandatory mixed type hints
+                if (!in_array($tokens[$typeHintPtr]['code'], [T_STRING, T_ARRAY_HINT])) {
+                    $error = sprintf(
+                        'Expected Type hint for the parameter "%1$s" in method "%2$s::%3$s"',
+                        $tokens[$i]['content'],
+                        $className,
+                        $methodName
+                    );
+                    $file->addError($error, $typeHintPtr, 'MissingTypeHint');
+
+                    continue;
+                }
+
+                // Check for magic methods specific type hints
+                if (isset(self::$whitelist[$methodName]) && self::$whitelist[$methodName][0] !== false) {
+                    if (isset(self::$whitelist[$methodName][0][$paramCount])
+                        && self::$whitelist[$methodName][0][$paramCount] !== $tokens[$typeHintPtr]['content']
+                        && self::$whitelist[$methodName][0][$paramCount] !== false
+                    ) {
+                        $error = sprintf(
+                            'Expected type hint "%1$s" for parameter "%2$s" found "%3$s" for ',
+                            self::$whitelist[$methodName][0][$paramCount],
+                            $tokens[$i]['content'],
+                            $tokens[$typeHintPtr]['content']
+                        );
+                        $error .= sprintf('the magic method "%1$s::%2$s"', $className, $methodName);
+                        $file->addError($error, $typeHintPtr, 'WrongTypeHint');
+
+                        continue;
+                    }
+                }
+
+                // If array type hint, enforce more specific documentation at @param
+                if ($tokens[$typeHintPtr]['content'] === 'array') {
+                    if (!$this->ifParamDocExistForArrayParameter($tokens[$i]['content'], $paramDoc)) {
+                        $error = sprintf(
+                            'Array type hint for the parameter "%1$s" in method "%2$s::%3$s" must be documented to ',
+                            $tokens[$i]['content'],
+                            $className,
+                            $methodName
+                        );
+                        $error .= sprintf(
```

Comment:
Can be a single sprintf() statement using position parameters (`%1$s`, `%2$s` and so on)